fdbclient/ActorLineageProfiler.cpp

```diff
@@ -201,3 +194,86 @@ std::shared_ptr<Sample> SampleCollectorT::collect() {
 	packer.pack(res);
 	return packer.done(time);
 }
+
+void SampleCollection_t::refresh() {
+	auto sample = _collector->collect();
+	auto min = sample->time - windowSize;
+	double oldest = 0.0;
+	while (oldest < min && !data.empty()) {
+		// we remove at most 10 elements at a time. This is so we don't block the main thread for too long.
+		{
+			Lock _{ mutex };
+			int i = 0;
+			do {
+				oldest = data.front()->time;
+				data.pop_front();
+				++i;
+			} while (i < 10 && oldest < min && !data.empty());
+		}
+	}
+	{
+		Lock _{ mutex };
+		data.push_back(sample);
+	}
+}
+
+std::vector<std::shared_ptr<Sample>> SampleCollection_t::get(double from /*= 0.0*/,
+                                                             double to /*= std::numeric_limits<double>::max()*/) const {
+	Lock _{ mutex };
+	std::vector<std::shared_ptr<Sample>> res;
+	for (const auto& sample : data) {
+		if (sample->time > to) {
+			break;
+		} else if (sample->time > from) {
+			res.emplace_back(sample);
```

Comment:
This should be `push_back` right? `sample` is an lvalue and this will still invoke the copy constructor, but it's just a shared_ptr anyway so it doesn't matter as much. Just a little clearer to use `push_back` here I think.